### ISSUE TITLE
Training results not good as expected 

### ISSUE THREAD

**Issue by pigtigger** [Vegeta2020/SE-SSD#35]
hi @Vegeta2020 , SE-SSD is a wonderful work ! but my results is so low after training about 50 epoch, I  only change the data path and set load_from to the model provided in README , could you give me some advice? 
Here is my results after training 50 epochs
![image](https://user-images.githubusercontent.com/39974202/129694019-ca583a76-3472-4c3f-b294-0374869d7b4f.png)

**Comment by Vegeta2020** [comment]
I guess you used the info / dbinfo pkl files or gt_database produced by other repositories? Otherwise, it's hard to get such an result.

**Comment by pigtigger** [comment]
Thanks @Vegeta2020 
I use create_data.py in this repo to create gt_database. 
I also try to use CIA-SSD pretrained model and se-ssd model trained from scrath to initialize teacher and student, but both results are not good. Could you share your initialization model? I assume the initialzation model is the keypoint.

**Comment by pigtigger** [comment]
@Vegeta2020 the loss during training gets larger and larger and I cant get a good result (like result in README, there is a 20 mAP gap). it block me several days :(

**Comment by AndyYuan96** [comment]
> @Vegeta2020 the loss during training gets larger and larger and I cant get a good result (like result in README, there is a 20 mAP gap). it block me several days :(

Hi, pigtigger, did you reproduce the paper result?

**Comment by pigtigger** [comment]
> > @Vegeta2020 the loss during training gets larger and larger and I cant get a good result (like result in README, there is a 20 mAP gap). it block me several days :(
> 
> Hi, pigtigger, did you reproduce the paper result?

not yet

**Comment by AndyYuan96** [comment]
> > > @Vegeta2020 the loss during training gets larger and larger and I cant get a good result (like result in README, there is a 20 mAP gap). it block me several days :(
> > 
> > 
> > Hi, pigtigger, did you reproduce the paper result?
> 
> not yet

so sad😭

**Comment by AndyYuan96** [comment]
> > > @Vegeta2020 the loss during training gets larger and larger and I cant get a good result (like result in README, there is a 20 mAP gap). it block me several days :(
> > 
> > 
> > Hi, pigtigger, did you reproduce the paper result?
> 
> not yet

hi, pigtigger, now I can finally run the code, but I can set batch size = 1 with a v100 32 gpu, does it normal ? when set batch_size = 2, error happen, CUDNN_STATUS_NOT_SUPPORTED. This error may appear if you passed in a  non-contiguous input. 
when set batch size = 1, the actually memory occupied is 12G.

**Comment by AndyYuan96** [comment]
> > > @Vegeta2020 the loss during training gets larger and larger and I cant get a good result (like result in README, there is a 20 mAP gap). it block me several days :(
> > 
> > 
> > Hi, pigtigger, did you reproduce the paper result?
> 
> not yet

After training after 13 epochs， my loss looks like increasing。。so sad

**Comment by pigtigger** [comment]
I meet the same problem, does anyone reproduce the results?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Vegeta2020/SE-SSD"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月9日(星期四) 晚上7:41
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Vegeta2020/SE-SSD] Training results not good as expected  (#35)





    
@Vegeta2020 the loss during training gets larger and larger and I cant get a good result (like result in README, there is a 20 mAP gap). it block me several days :(
  
Hi, pigtigger, did you reproduce the paper result?
  
not yet
  
After training after 13 epochs， my loss looks like increasing。。so sad

**Comment by AndyYuan96** [comment]
> I meet the same problem, does anyone reproduce the results?
> […](#)

I just wonder how much gpu memory you need when reproducing, I can only set batch_size = 1 even I use a v100 32g gpu

**Comment by pigtigger** [comment]
I set 4 on 24G Titan, it's ok.
I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Vegeta2020/SE-SSD"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月9日(星期四) 晚上8:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Vegeta2020/SE-SSD] Training results not good as expected  (#35)





  
I meet the same problem, does anyone reproduce the results?
 …
  
I just wonder how much gpu memory you need when reproducing, I can only set batch_size = 1 even I use a v100 32g gpu

**Comment by AndyYuan96** [comment]
> I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?
> […](#)

I did use pretrain model, in code I just change model_ema = build_detector(), no other change, when I set batch_size = 2, just error

**Comment by AndyYuan96** [comment]
> I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?
> […](#)

my initial loss is 0.55 around.

**Comment by AndyYuan96** [comment]
> I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?
> […](#)

what's more, I'm new to det3d, I didn't see the tensorboard log, did you have tensorboard log when reproducing the code.

**Comment by pigtigger** [comment]
> > I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?
> > […](#)
> 
> my initial loss is 0.55 around.

which pretrained model did you use ? I have tried CIA-pretrained model and the model provided in SE-SSD README, the initial loss is about 0.55 for CIA-pretrained model  and  0.45 for SE-SSD, but both  can not get a good result

**Comment by AndyYuan96** [comment]
> > > I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?
> > > […](#)
> > 
> > 
> > my initial loss is 0.55 around.
> 
> which pretrained model did you use ? I have tried CIA-pretrained model and the model provided in SE-SSD README, the initial loss is about 0.55 for CIA-pretrained model and 0.45 for SE-SSD, but both can not get a good result

CIA-pretrained, as I see the author say use this one in other question

**Comment by AndyYuan96** [comment]
> > > I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?
> > > […](#)
> > 
> > 
> > my initial loss is 0.55 around.
> 
> which pretrained model did you use ? I have tried CIA-pretrained model and the model provided in SE-SSD README, the initial loss is about 0.55 for CIA-pretrained model and 0.45 for SE-SSD, but both can not get a good result

It's a little strange why I can only set batch size = 1 with v100 32g

**Comment by pigtigger** [comment]
I did not meet the gpu memory error after using pretrained model, it is so strange , maybe&nbsp; it is&nbsp; env problem since you did not change the code, btw&nbsp; I also change the line you mentioned before&nbsp;
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Vegeta2020/SE-SSD"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月9日(星期四) 晚上8:35
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Vegeta2020/SE-SSD] Training results not good as expected  (#35)





    
I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that?
 …
  
my initial loss is 0.55 around.
  
which pretrained model did you use ? I have tried CIA-pretrained model and the model provided in SE-SSD README, the initial loss is about 0.55 for CIA-pretrained model and 0.45 for SE-SSD, but both can not get a good result
  
It's a little strange why I can only set batch size = 1 with v100 32g

**Comment by AndyYuan96** [comment]
> I did not meet the gpu memory error after using pretrained model, it is so strange , maybe&nbsp; it is&nbsp; env problem since you did not change the code, btw&nbsp; I also change the line you mentioned before&nbsp; ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Vegeta2020/SE-SSD" ***@***.***&gt;; 发送时间:&nbsp;2021年9月9日(星期四) 晚上8:35 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [Vegeta2020/SE-SSD] Training results not good as expected (#35) I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that? … my initial loss is 0.55 around. which pretrained model did you use ? I have tried CIA-pretrained model and the model provided in SE-SSD README, the initial loss is about 0.55 for CIA-pretrained model and 0.45 for SE-SSD, but both can not get a good result It's a little strange why I can only set batch size = 1 with v100 32g — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

tomorrow morning will finish training, looking forward to the result.

**Comment by AndyYuan96** [comment]
> I did not meet the gpu memory error after using pretrained model, it is so strange , maybe&nbsp; it is&nbsp; env problem since you did not change the code, btw&nbsp; I also change the line you mentioned before&nbsp; ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Vegeta2020/SE-SSD" ***@***.***&gt;; 发送时间:&nbsp;2021年9月9日(星期四) 晚上8:35 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [Vegeta2020/SE-SSD] Training results not good as expected (#35) I set 4 on 24G Titan, it's ok. I guess you need to use a pretrained model to initialize the student and teacher model, have you tried that? … my initial loss is 0.55 around. which pretrained model did you use ? I have tried CIA-pretrained model and the model provided in SE-SSD README, the initial loss is about 0.55 for CIA-pretrained model and 0.45 for SE-SSD, but both can not get a good result It's a little strange why I can only set batch size = 1 with v100 32g — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

So sad, after SE-SSD training, don't have improvement compared with the pretrained model provided by CIA-SSD repo.

**Comment by pigtigger** [comment]
@AndyYuan96 did you train the CIA-SSD pretrained model yourself?

**Comment by AndyYuan96** [comment]
> @AndyYuan96 did you train the CIA-SSD pretrained model yourself?

no，I use the model from CIA-SSD repo as pretrained model.
